### ISSUE TITLE
Bad size of marks file (19.14.6.12)

### ISSUE THREAD

**Issue by reflection** [ClickHouse/ClickHouse#7067]
```
2019.09.23 18:44:18.891768 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> executeQuery: (from [::ffff:172.17.0.8]:49798) SELECT CAST(formatDateTime(toStartOfHour(timestamp, 'America/New_York'), '%H', 'America/New_York') AS Nullable(String)) AS hour, CAST(arrayJoin(category) AS Nullable(String)) AS category, domain AS domain, "user" AS "user", count(timestamp) AS requests, sum(size) AS bandwidth  FROM wsg_wf  WHERE hour IS NOT NULL AND category IS NOT NULL AND domain IS NOT NULL AND "user" IS NOT NULL AND timestamp >= 1568606400 AND timestamp <= 1569261600 AND sn IN ('974599', '1095961', '1001616', '980201', '1001703', '972439', '1001662', '974366', '1095974', '1095903', '971821', '988292', '1092793', '971775', '970505', '1054741', '970507', '1089171', '960522', '1004933', '1066554', '1064055', '1095970', '960399', '974476', '950654', '971803', '1005107', '988312', '1001618', '1001119', '1089481', '975954', '1065667', '1001719', '971694', '1004936', '1001720', '1004932', '1095902', '960359', '971816', '1065665', '1095953', '950660', '1004935', '970527', '972396', '971807', '989933', '971605', '1095857', '972130', '972395', '960212', '972361', '1095904', '960213', '974665', '1001702', '980206') GROUP BY hour, category, domain, "user" WITH ROLLUP ORDER BY hour ASC FORMAT TabSeparatedWithNamesAndTypes
2019.09.23 18:44:18.898150 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> InterpreterSelectQuery: MergeTreeWhereOptimizer: condition "isNotNull(CAST(formatDateTime(toStartOfHour(timestamp, 'America/New_York'), '%H', 'America/New_York'), 'Nullable(String)') AS hour) AND (sn IN ('974599', '1095961', '1001616', '980201', '1001703', '972439', '1001662', '974366', '1095974', '1095903', '971821', '988292', '1092793', '971775', '970505', '1054741', '970507', '1089171', '960522', '1004933', '1066554', '1064055', '1095970', '960399', '974476', '950654', '971803', '1005107', '988312', '1001618', '1001119', '1089481', '975954', '1065667', '1001719', '971694', '1004936', '1001720', '1004932', '1095902', '960359', '971816', '1065665', '1095953', '950660', '1004935', '970527', '972396', '971807', '989933', '971605', '1095857', '972130', '972395', '960212', '972361', '1095904', '960213', '974665', '1001702', '980206'))" moved to PREWHERE
2019.09.23 18:44:18.900206 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Key condition: unknown, unknown, and, unknown, and, (column 0 in [1568606400, +inf)), and, (column 0 in (-inf, 1569261600]), and, unknown, unknown, and, and
2019.09.23 18:44:18.900240 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): MinMax index condition: unknown, unknown, and, unknown, and, (column 0 in [1568606400, +inf)), and, (column 0 in (-inf, 1569261600]), and, unknown, unknown, and, and
2019.09.23 18:44:18.906319 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `sn_idx` has dropped 0 granules.
2019.09.23 18:44:18.910097 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `category_idx` has dropped 0 granules.
2019.09.23 18:44:18.911307 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `domain_idx` has dropped 0 granules.
2019.09.23 18:44:18.915887 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `sn_idx` has dropped 0 granules.
2019.09.23 18:44:18.919891 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `category_idx` has dropped 0 granules.
2019.09.23 18:44:18.922159 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `domain_idx` has dropped 0 granules.
2019.09.23 18:44:18.923946 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `sn_idx` has dropped 0 granules.
2019.09.23 18:44:18.926963 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `category_idx` has dropped 0 granules.
2019.09.23 18:44:18.928368 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `domain_idx` has dropped 0 granules.
2019.09.23 18:44:18.928675 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `sn_idx` has dropped 0 granules.
2019.09.23 18:44:18.930614 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `category_idx` has dropped 0 granules.
2019.09.23 18:44:18.931476 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `domain_idx` has dropped 0 granules.
2019.09.23 18:44:18.931828 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `sn_idx` has dropped 0 granules.
2019.09.23 18:44:18.934924 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `category_idx` has dropped 0 granules.
2019.09.23 18:44:18.935497 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `domain_idx` has dropped 0 granules.
2019.09.23 18:44:18.935858 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `sn_idx` has dropped 0 granules.
2019.09.23 18:44:18.937739 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `category_idx` has dropped 0 granules.
2019.09.23 18:44:18.938349 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `domain_idx` has dropped 0 granules.
2019.09.23 18:44:18.938574 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `sn_idx` has dropped 0 granules.
2019.09.23 18:44:18.940429 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `category_idx` has dropped 0 granules.
2019.09.23 18:44:18.940903 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `domain_idx` has dropped 0 granules.
2019.09.23 18:44:18.941094 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `sn_idx` has dropped 0 granules.
2019.09.23 18:44:18.942326 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `category_idx` has dropped 0 granules.
2019.09.23 18:44:18.942663 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `domain_idx` has dropped 0 granules.
2019.09.23 18:44:18.942796 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `sn_idx` has dropped 0 granules.
2019.09.23 18:44:18.943133 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `category_idx` has dropped 0 granules.
2019.09.23 18:44:18.943260 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `domain_idx` has dropped 0 granules.
2019.09.23 18:44:18.943384 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `sn_idx` has dropped 0 granules.
2019.09.23 18:44:18.943489 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `category_idx` has dropped 0 granules.
2019.09.23 18:44:18.943580 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `domain_idx` has dropped 0 granules.
2019.09.23 18:44:18.949158 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `sn_idx` has dropped 0 granules.
2019.09.23 18:44:18.952468 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `category_idx` has dropped 0 granules.
2019.09.23 18:44:18.953575 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `domain_idx` has dropped 0 granules.
2019.09.23 18:44:18.957263 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `sn_idx` has dropped 0 granules.
2019.09.23 18:44:18.959753 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `category_idx` has dropped 0 granules.
2019.09.23 18:44:18.960821 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `domain_idx` has dropped 0 granules.
2019.09.23 18:44:18.961525 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `sn_idx` has dropped 0 granules.
2019.09.23 18:44:18.963577 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `category_idx` has dropped 0 granules.
2019.09.23 18:44:18.964255 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> default.wsg_wf (SelectExecutor): Index `domain_idx` has dropped 0 granules.
2019.09.23 18:44:18.965500 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> MemoryTracker: Peak memory usage (total): 9.18 MiB.
2019.09.23 18:44:18.965584 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Error> executeQuery: Code: 246, e.displayText() = DB::Exception: Bad size of marks file '/var/lib/clickhouse/data/default/wsg_wf/20190921_137626_143502_28/skp_idx_sn_idx.mrk2': 48, must be: 24 (version 19.14.6.12 (official build)) (from [::ffff:172.17.0.8]:49798) (in query: SELECT CAST(formatDateTime(toStartOfHour(timestamp, 'America/New_York'), '%H', 'America/New_York') AS Nullable(String)) AS hour, CAST(arrayJoin(category) AS Nullable(String)) AS category, domain AS domain, "user" AS "user", count(timestamp) AS requests, sum(size) AS bandwidth  FROM wsg_wf  WHERE hour IS NOT NULL AND category IS NOT NULL AND domain IS NOT NULL AND "user" IS NOT NULL AND timestamp >= 1568606400 AND timestamp <= 1569261600 AND sn IN ('974599', '1095961', '1001616', '980201', '1001703', '972439', '1001662', '974366', '1095974', '1095903', '971821', '988292', '1092793', '971775', '970505', '1054741', '970507', '1089171', '960522', '1004933', '1066554', '1064055', '1095970', '960399', '974476', '950654', '971803', '1005107', '988312', '1001618', '1001119', '1089481', '975954', '1065667', '1001719', '971694', '1004936', '1001720', '1004932', '1095902', '960359', '971816', '1065665', '1095953', '950660', '1004935', '970527', '972396', '971807', '989933', '971605', '1095857', '972130', '972395', '960212', '972361', '1095904', '960213', '974665', '1001702', '980206') GROUP BY hour, category, domain, "user" WITH ROLLUP ORDER BY hour ASC FORMAT TabSeparatedWithNamesAndTypes), Stack trace:

0. 0x55834aeb3160 StackTrace::StackTrace() /usr/bin/clickhouse
1. 0x55834aeb2f35 DB::Exception::Exception(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int) /usr/bin/clickhouse
2. 0x55834acadffb ? /usr/bin/clickhouse
3. 0x55834e4aa42a DB::MergeTreeReaderStream::loadMarks() /usr/bin/clickhouse
4. 0x55834e4ab858 DB::MergeTreeReaderStream::MergeTreeReaderStream(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned long, std::vector<DB::MarkRange, std::allocator<DB::MarkRange> > const&, DB::MarkCache*, bool, DB::UncompressedCache*, unsigned long, unsigned long, unsigned long, DB::MergeTreeIndexGranularityInfo const*, std::function<void (DB::ReadBufferFromFileBase::ProfileInfo)> const&, int) /usr/bin/clickhouse
5. 0x55834e49b05b DB::MergeTreeIndexReader::MergeTreeIndexReader(std::shared_ptr<DB::IMergeTreeIndex const>, std::shared_ptr<DB::MergeTreeDataPart const>, unsigned long, std::vector<DB::MarkRange, std::allocator<DB::MarkRange> > const&) /usr/bin/clickhouse
6. 0x55834e487d2e DB::MergeTreeDataSelectExecutor::filterMarksUsingIndex(std::shared_ptr<DB::IMergeTreeIndex const>, std::shared_ptr<DB::IMergeTreeIndexCondition>, std::shared_ptr<DB::MergeTreeDataPart const>, std::vector<DB::MarkRange, std::allocator<DB::MarkRange> > const&, DB::Settings const&) const /usr/bin/clickhouse
7. 0x55834e48a081 DB::MergeTreeDataSelectExecutor::readFromParts(std::vector<std::shared_ptr<DB::MergeTreeDataPart const>, std::allocator<std::shared_ptr<DB::MergeTreeDataPart const> > >, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, DB::SelectQueryInfo const&, DB::Context const&, unsigned long, unsigned int, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, long, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, long> > > const*) const /usr/bin/clickhouse
8. 0x55834e48d419 DB::MergeTreeDataSelectExecutor::read(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, DB::SelectQueryInfo const&, DB::Context const&, unsigned long, unsigned int, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, long, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, long> > > const*) const /usr/bin/clickhouse
9. 0x55834e37a7d2 DB::StorageMergeTree::read(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, DB::SelectQueryInfo const&, DB::Context const&, DB::QueryProcessingStage::Enum, unsigned long, unsigned int) /usr/bin/clickhouse
10. 0x55834e163f9f void DB::InterpreterSelectQuery::executeFetchColumns<DB::InterpreterSelectQuery::Pipeline>(DB::QueryProcessingStage::Enum, DB::InterpreterSelectQuery::Pipeline&, std::shared_ptr<DB::SortingInfo> const&, std::shared_ptr<DB::PrewhereInfo> const&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&) /usr/bin/clickhouse
11. 0x55834e16a5de void DB::InterpreterSelectQuery::executeImpl<DB::InterpreterSelectQuery::Pipeline>(DB::InterpreterSelectQuery::Pipeline&, std::shared_ptr<DB::IBlockInputStream> const&) /usr/bin/clickhouse
12. 0x55834e14d535 DB::InterpreterSelectQuery::executeWithMultipleStreams() /usr/bin/clickhouse
13. 0x55834e16c303 DB::InterpreterSelectWithUnionQuery::executeWithMultipleStreams() /usr/bin/clickhouse
14. 0x55834e16e1c8 DB::InterpreterSelectWithUnionQuery::execute() /usr/bin/clickhouse
15. 0x55834e27b680 ? /usr/bin/clickhouse
16. 0x55834e27cb2c DB::executeQuery(DB::ReadBuffer&, DB::WriteBuffer&, bool, DB::Context&, std::function<void (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)>, std::function<void (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)>) /usr/bin/clickhouse
17. 0x55834af33e1e DB::HTTPHandler::processQuery(Poco::Net::HTTPServerRequest&, HTMLForm&, Poco::Net::HTTPServerResponse&, DB::HTTPHandler::Output&) /usr/bin/clickhouse
18. 0x55834af361c9 DB::HTTPHandler::handleRequest(Poco::Net::HTTPServerRequest&, Poco::Net::HTTPServerResponse&) /usr/bin/clickhouse
19. 0x55834ed9eda9 Poco::Net::HTTPServerConnection::run() /usr/bin/clickhouse
20. 0x55834ed99b80 Poco::Net::TCPServerConnection::start() /usr/bin/clickhouse
21. 0x55834ed9a29d Poco::Net::TCPServerDispatcher::run() /usr/bin/clickhouse
22. 0x5583503e8b31 Poco::PooledThread::run() /usr/bin/clickhouse
23. 0x5583503e68dc Poco::ThreadImpl::runnableEntry(void*) /usr/bin/clickhouse
24. 0x558350b5c120 ? /usr/bin/clickhouse
25. 0x7fdb631a84a4 start_thread /lib/x86_64-linux-gnu/libpthread-2.24.so
26. 0x7fdb62aded0f clone /lib/x86_64-linux-gnu/libc-2.24.so

2019.09.23 18:44:18.965723 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Debug> MemoryTracker: Peak memory usage (for query): 9.18 MiB.
2019.09.23 18:44:18.965771 [ 55 ] {269a43e6-de32-11e9-bb04-0242ac110008} <Error> HTTPHandler: Code: 246, e.displayText() = DB::Exception: Bad size of marks file '/var/lib/clickhouse/data/default/wsg_wf/20190921_137626_143502_28/skp_idx_sn_idx.mrk2': 48, must be: 24, Stack trace:

0. 0x55834aeb3160 StackTrace::StackTrace() /usr/bin/clickhouse
1. 0x55834aeb2f35 DB::Exception::Exception(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int) /usr/bin/clickhouse
2. 0x55834acadffb ? /usr/bin/clickhouse
3. 0x55834e4aa42a DB::MergeTreeReaderStream::loadMarks() /usr/bin/clickhouse
4. 0x55834e4ab858 DB::MergeTreeReaderStream::MergeTreeReaderStream(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned long, std::vector<DB::MarkRange, std::allocator<DB::MarkRange> > const&, DB::MarkCache*, bool, DB::UncompressedCache*, unsigned long, unsigned long, unsigned long, DB::MergeTreeIndexGranularityInfo const*, std::function<void (DB::ReadBufferFromFileBase::ProfileInfo)> const&, int) /usr/bin/clickhouse
5. 0x55834e49b05b DB::MergeTreeIndexReader::MergeTreeIndexReader(std::shared_ptr<DB::IMergeTreeIndex const>, std::shared_ptr<DB::MergeTreeDataPart const>, unsigned long, std::vector<DB::MarkRange, std::allocator<DB::MarkRange> > const&) /usr/bin/clickhouse
6. 0x55834e487d2e DB::MergeTreeDataSelectExecutor::filterMarksUsingIndex(std::shared_ptr<DB::IMergeTreeIndex const>, std::shared_ptr<DB::IMergeTreeIndexCondition>, std::shared_ptr<DB::MergeTreeDataPart const>, std::vector<DB::MarkRange, std::allocator<DB::MarkRange> > const&, DB::Settings const&) const /usr/bin/clickhouse
7. 0x55834e48a081 DB::MergeTreeDataSelectExecutor::readFromParts(std::vector<std::shared_ptr<DB::MergeTreeDataPart const>, std::allocator<std::shared_ptr<DB::MergeTreeDataPart const> > >, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, DB::SelectQueryInfo const&, DB::Context const&, unsigned long, unsigned int, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, long, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, long> > > const*) const /usr/bin/clickhouse
8. 0x55834e48d419 DB::MergeTreeDataSelectExecutor::read(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, DB::SelectQueryInfo const&, DB::Context const&, unsigned long, unsigned int, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, long, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, long> > > const*) const /usr/bin/clickhouse
9. 0x55834e37a7d2 DB::StorageMergeTree::read(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, DB::SelectQueryInfo const&, DB::Context const&, DB::QueryProcessingStage::Enum, unsigned long, unsigned int) /usr/bin/clickhouse
10. 0x55834e163f9f void DB::InterpreterSelectQuery::executeFetchColumns<DB::InterpreterSelectQuery::Pipeline>(DB::QueryProcessingStage::Enum, DB::InterpreterSelectQuery::Pipeline&, std::shared_ptr<DB::SortingInfo> const&, std::shared_ptr<DB::PrewhereInfo> const&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&) /usr/bin/clickhouse
11. 0x55834e16a5de void DB::InterpreterSelectQuery::executeImpl<DB::InterpreterSelectQuery::Pipeline>(DB::InterpreterSelectQuery::Pipeline&, std::shared_ptr<DB::IBlockInputStream> const&) /usr/bin/clickhouse
12. 0x55834e14d535 DB::InterpreterSelectQuery::executeWithMultipleStreams() /usr/bin/clickhouse
13. 0x55834e16c303 DB::InterpreterSelectWithUnionQuery::executeWithMultipleStreams() /usr/bin/clickhouse
14. 0x55834e16e1c8 DB::InterpreterSelectWithUnionQuery::execute() /usr/bin/clickhouse
15. 0x55834e27b680 ? /usr/bin/clickhouse
16. 0x55834e27cb2c DB::executeQuery(DB::ReadBuffer&, DB::WriteBuffer&, bool, DB::Context&, std::function<void (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)>, std::function<void (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)>) /usr/bin/clickhouse
17. 0x55834af33e1e DB::HTTPHandler::processQuery(Poco::Net::HTTPServerRequest&, HTMLForm&, Poco::Net::HTTPServerResponse&, DB::HTTPHandler::Output&) /usr/bin/clickhouse
18. 0x55834af361c9 DB::HTTPHandler::handleRequest(Poco::Net::HTTPServerRequest&, Poco::Net::HTTPServerResponse&) /usr/bin/clickhouse
19. 0x55834ed9eda9 Poco::Net::HTTPServerConnection::run() /usr/bin/clickhouse
20. 0x55834ed99b80 Poco::Net::TCPServerConnection::start() /usr/bin/clickhouse
21. 0x55834ed9a29d Poco::Net::TCPServerDispatcher::run() /usr/bin/clickhouse
22. 0x5583503e8b31 Poco::PooledThread::run() /usr/bin/clickhouse
23. 0x5583503e68dc Poco::ThreadImpl::runnableEntry(void*) /usr/bin/clickhouse
24. 0x558350b5c120 ? /usr/bin/clickhouse
25. 0x7fdb631a84a4 start_thread /lib/x86_64-linux-gnu/libpthread-2.24.so
26. 0x7fdb62aded0f clone /lib/x86_64-linux-gnu/libc-2.24.so
 (version 19.14.6.12 (official build))
```

**Comment by alexey-milovidov** [comment]
Could you please share your table definition?
(The most relevant parts are INDEX definitions)

**Comment by reflection** [comment]
```
CREATE TABLE IF NOT EXISTS wsg.wsg_wf (
  timestamp         DateTime,
  sn                LowCardinality(Nullable(String)),
  action            Nullable(UInt8),
  app_rule          LowCardinality(Nullable(String)),
  category          Array(LowCardinality(Nullable(String))),
  content_type      LowCardinality(Nullable(String)),
  dst_ip            Nullable(UInt32),
  dst_ip_country    LowCardinality(Nullable(String)),
  dst_ip_lat        Nullable(Decimal(18,16)),
  dst_ip_lon        Nullable(Decimal(19,16)),
  dst_port          Nullable(UInt16),
  domain            LowCardinality(Nullable(String)),
  fw_rule           LowCardinality(Nullable(String)),
  method            LowCardinality(Nullable(String)),
  name              LowCardinality(Nullable(String)),
  size              Nullable(UInt64),
  super_category    Array(LowCardinality(Nullable(String))),
  src_ip            Nullable(UInt32),
  src_ip_country    LowCardinality(Nullable(String)),
  src_ip_lat        Nullable(Decimal(18,16)),
  src_ip_lon        Nullable(Decimal(19,16)),
  src_port          Nullable(UInt16),
  status_code       Nullable(UInt16),
  traffic_type      Nullable(UInt8),
  user              LowCardinality(Nullable(String)),
  user_agent        LowCardinality(Nullable(String)),
  user_type         Nullable(UInt8),
  url               LowCardinality(Nullable(String)),
  INDEX sn_idx (sn) TYPE set(0) GRANULARITY 8192,
  INDEX action_idx (action) TYPE set(0) GRANULARITY 8192,
  INDEX category_idx (category) TYPE set(0) GRANULARITY 8192,
  INDEX domain_idx (domain) TYPE set(0) GRANULARITY 8192,
  INDEX super_category_idx (super_category) TYPE set(0) GRANULARITY 8192,
  INDEX src_ip_idx (src_ip) TYPE set(0) GRANULARITY 8192,
  INDEX user_idx (user) TYPE set(0) GRANULARITY 8192
) ENGINE=MergeTree
  PARTITION BY toYYYYMMDD(timestamp)
  ORDER BY (timestamp, intHash64(timestamp))
  SAMPLE BY intHash64(timestamp);
```

**Comment by reflection** [comment]
For context, we saw the `bad size of marks` error with the 19.13.2.19 build, so we tried updating to the newer 19.14.6.12 build, but still saw the same error. 

We tried a few things (and still have the old data if needed to help y'all with debugging):

- Tried recreating indices, but were getting errors so had to kill mutation:
```
┌─database─┬─table──┬─mutation_id─────────┬─command───────────────────────────────────────────────────────────┬─────────create_time─┬─block_numbers.partition_id─┬─block_numbers.number─┬─parts_to_do─┬─is_done─┬─latest_failed_part──┬────latest_fail_time─┬─latest_fail_reason────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ default  │ wsg_wf │ mutation_187958.txt │ MATERIALIZE INDEX sn_idx IN PARTITION '20190921_150933_153087_27' │ 2019-09-23 20:40:32 │ ['']                       │ [187958]             │          28 │       0 │ 20190920_1_24387_30 │ 2019-09-23 20:47:15 │ Code: 53, e.displayText() = DB::Exception: Type mismatch in IN or VALUES section. Expected: UInt32. Got: String (version 19.14.6.12 (official build)) │
└──────────┴────────┴─────────────────────┴───────────────────────────────────────────────────────────────────┴─────────────────────┴────────────────────────────┴──────────────────────┴─────────────┴─────────┴─────────────────────┴─────────────────────┴───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘

2019.09.23 20:47:14.465219 [ 15 ] {} <Error> void DB::BackgroundProcessingPool::threadFunction(): Code: 53, e.displayText() = DB::Exception: Type mismatch in IN or VALUES section. Expected: UInt32. Got: String, Stack trace:

0. 0x5608b93c7160 StackTrace::StackTrace() /usr/bin/clickhouse
1. 0x5608b93c6f35 DB::Exception::Exception(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int) /usr/bin/clickhouse
2. 0x5608b9176273 ? /usr/bin/clickhouse
3. 0x5608bc7852ab DB::convertFieldToType(DB::Field const&, DB::IDataType const&, DB::IDataType const*) /usr/bin/clickhouse
4. 0x5608b91fa49b ? /usr/bin/clickhouse
5. 0x5608bceb7b1d DB::IRowInputFormat::generate() /usr/bin/clickhouse
6. 0x5608bca8ee8e DB::ISource::work() /usr/bin/clickhouse
7. 0x5608bca6a305 DB::InputStreamFromInputFormat::readImpl() /usr/bin/clickhouse
8. 0x5608bc500537 DB::IBlockInputStream::read() /usr/bin/clickhouse
9. 0x5608bc947f72 DB::MergeTreeData::getPartitionIDFromQuery[abi:cxx11](std::shared_ptr<DB::IAST> const&, DB::Context const&) /usr/bin/clickhouse
10. 0x5608bc971ff1 DB::MergeTreeDataMergerMutator::mutatePartToTemporaryPart(DB::FutureMergedMutatedPart const&, std::vector<DB::MutationCommand, std::allocator<DB::MutationCommand> > const&, DB::MergeListEntry&, DB::Context const&, DB::TableStructureReadLockHolder&) /usr/bin/clickhouse
11. 0x5608bc897beb DB::StorageMergeTree::tryMutatePart() /usr/bin/clickhouse
12. 0x5608bc898057 DB::StorageMergeTree::backgroundTask() /usr/bin/clickhouse
13. 0x5608bc9229e4 DB::BackgroundProcessingPool::threadFunction() /usr/bin/clickhouse
14. 0x5608bc92335a ? /usr/bin/clickhouse
15. 0x5608b940230c ThreadPoolImpl<std::thread>::worker(std::_List_iterator<std::thread>) /usr/bin/clickhouse
16. 0x5608bf070120 ? /usr/bin/clickhouse
17. 0x7fe027cae4a4 start_thread /lib/x86_64-linux-gnu/libpthread-2.24.so
18. 0x7fe0275e4d0f clone /lib/x86_64-linux-gnu/libc-2.24.so
 (version 19.14.6.12 (official build))
2019.09.23 20:47:14.609144 [ 20 ] {} <Error> void DB::BackgroundProcessingPool::threadFunction(): Code: 53, e.displayText() = DB::Exception: Type mismatch in IN or VALUES section. Expected: UInt32. Got: String, Stack trace:
```
- Started with a "fresh" table and inserted some rows.  No errors so far, but will update this issue tomorrow after more data is inserted.
- Still have old data if needed to help with debugging.  Would be awesome if CH fixed any older indices that might have issues, as we have many db deployments.  Thanks.

**Comment by alesapin** [comment]
Unfortunately, this bug was fixed exactly at 19.13.3: https://github.com/ClickHouse/ClickHouse/pull/6748. The only solution for old data -- remove indices and recreate them again. 

BTW, it seems like your indices have the wrong definition. Skip indices granularity specified in primary index granularity which by default already approximately one record per 8192 rows. So, your skip indices calculated for each 8192 * 8192 row, which is very rare and not useful (it doesn't speed up queries). 1,2,4 or 8 are recommended values for skip indices.
